zulip_bots/zulip_bots/lib.py

```diff
@@ -166,7 +166,7 @@ def send_reply(self, message: Dict[str, Any], response: str, widget_content: Opt
         if message['type'] == 'private':
             return self.send_message(dict(
                 type='private',
-                to=[x['email'] for x in message['display_recipient'] if self.email != x['email']],
```

Comment:
This was one of my very early contributions. Looking at it again now, my thought would be that it was intended to avoid a bot triggering itself with its own message, due to auto-mentions in PMs. Can that happen now, assuming it did before?

If so, this behavior is maybe countered by the other commit (though not fully?), so should these commits be squashed or restructured?

Reply:
Does this have to be paired with server logic like zulip/zulip#14241, to not break?

Reply:
No. This will work fine either way. Paired with the the server change, we address the case when the bot is IN the list of recipients but afterwards, the bot won't be (since the server won't automatically add the sender to recipients list).